src/Oscar.jl

```diff
@@ -401,40 +401,55 @@ end
 
 
 @doc raw"""
-    test_module(x, new::Bool = true)
+    test_module(file::AbstractString; new::Bool = true)
 
-Run the Oscar tests in the file `x.jl` where `x` is a string.
-
-If `x == "all"` run the entire testsuite.
+Run the Oscar tests in the file `test/<file>.jl` where `file` may be a path.
 
 The optional parameter `new` takes the values `false` and `true` (default). If
 `true`, then the tests are run in a new session, otherwise the currently active
 session is used.
+
+For experimental modules, use [`test_experimental_module`](@ref) instead.
 """
-function test_module(x::AbstractString, new::Bool = true)
-   julia_exe = Base.julia_cmd()
-   if x == "all"
-     test_file = joinpath(oscardir, "test/runtests.jl")
-   else
-     test_file = joinpath(oscardir, "test/$x.jl")
-   end
+function test_module(file::AbstractString; new::Bool=true)
+  julia_exe = Base.julia_cmd()
+  rel_test_file = normpath("test", "$file.jl")
+  test_file = joinpath(oscardir, rel_test_file)
+
+  if new
+    cmd = "using Test; using Oscar; Hecke.assertions(true); include(\"$test_file\");"
+    @info("spawning ", `$julia_exe -e \"$cmd\"`)
+    run(`$julia_exe -e $cmd`)
+  else
+    @info("Running tests for $rel_test_file in same session")
+    try
+      include(test_file)
+    catch e
+      if isa(e, LoadError)
+        println("You need to do \"using Test\"")
```

Comment:
Note a change request (but perhaps someone would like to follow this up in another PR or an issue): I don't like this error message. It can be widely misleading.

Instead of *guessing* that perhaps `using Test` is missing, we can do much better: We could *first* check whether it is available, even before the `include`, and refuse to even do the `include` if it isn't. 

My second concern is: we are `include`ing the test file into the Oscar namespace. But I don't think that's how tests are run when one uses `Pkg.test` and thus also not how CI runs it; and also not how this function does it when `new` is specified. Rather, the `include` is always done from the REPL.

So perhaps instead of `include(test_file)` we should be doing `Base.include(Base.Main, test_file)` ?